### PR TITLE
[release-0.5] backport CI fixes

### PR DESCRIPTION
**Pull request by vtjnash**
to try making travis happier

### CODE REVIEW

**Comment by maleadt** [comment]
Now that you're backporting CI fixes to 0.5, maybe include #20430? Example [CI failure](https://ci.maleadt.net/buildbot/julia/builders/Julia%200.5%20%28x86-64%29/builds/12).

Rebased patch:
```patch
From 7ff704c8feb084fad062f3e9a014dfc1c7d85c95 Mon Sep 17 00:00:00 2001
From: Tim Besard <tim.besard@gmail.com>
Date: Tue, 7 Feb 2017 13:01:27 +0100
Subject: [PATCH] Make sure flisp.boot is copied to an existing directory.

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index d7ec864f27..9aae087541 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -151,7 +151,7 @@ $(BUILDDIR)/julia_flisp.boot.inc: $(BUILDDIR)/julia_flisp.boot $(FLISP_EXECUTABL
 
 $(BUILDDIR)/julia_flisp.boot: $(addprefix $(SRCDIR)/,jlfrontend.scm \
                julia-parser.scm julia-syntax.scm match.scm utils.scm ast.scm macroexpand.scm mk_julia_flisp_boot.scm) \
-               $(FLISP_EXECUTABLE_release)
+               $(FLISP_EXECUTABLE_release) | $(BUILDDIR)
        @$(call PRINT_FLISP, $(call spawn,$(FLISP_EXECUTABLE_release)) \
                $(call cygpath_w,$(SRCDIR)/mk_julia_flisp_boot.scm) $(call cygpath_w,$(dir $<)) $(notdir $<) $(call cygpath_w,$@))
 
-- 
2.11.1
```

**Review by tkelman — CHANGES_REQUESTED** [review]
We should not be changing the libgit2 version on release-0.5

File: base/libgit2/types.jl [hunk 1]
```diff
@@ -174,7 +174,50 @@ function RemoteCallbacks(credentials::Ptr{Void}, payload::Ref{Nullable{AbstractC
     RemoteCallbacks(credentials=credentials_cb(), payload=pointer_from_objref(payload))
 end
 
-if LibGit2.version() >= v"0.24.0"
+immutable ProxyOptions
```
Comment (tkelman) [inline base/libgit2/types.jl]: should put the definition of this type under the `if LibGit2.version() >= v"0.25.0"` as well, it shouldn't be added to the module unless you're building in a non-default configuration and absolutely need it

Comment (vtjnash) [inline base/libgit2/types.jl]: It's not an API issue. I'm just attempting to roughly match whatever we have on master, not actually redesign that commit.

Comment (tkelman) [inline base/libgit2/types.jl]: release-0.5 shouldn't match what's on master, it should remain close to 0.5.0. just change msys_build.sh to download 0.5-latest and you won't need to touch libgit2 at all

Comment (vtjnash) [inline base/libgit2/types.jl]: I don't think semver cares about internal names. This is non-breaking at the API level to keep this slightly closer to master.

Comment (tkelman) [inline base/libgit2/types.jl]: I don't want to be defining an unused type in the default build configuration.

**Comment by StefanKarpinski** [comment]
Can we make some progress on this, pretty please?

**Review by tkelman — APPROVED** [review]

**Pull request merged**